mteb/evaluation/MTEB.py

```diff
@@ -11,6 +11,7 @@
 from typing import List, Union
 
 import datasets
+from codecarbon import EmissionsTracker
```

Comment:
Should we make the import optional, too?
I.e. moving `from codecarbon import EmissionsTracker` below `if co2_tracker:`?
So people aren't forced to install the package

Reply:
True, updating thanks!

Reply:
I think I should also update the pyptoject.toml, I initially put it in the dependencies so it gets installed with mteb. I don't know where it should go 🤔 

I don't think it should go here since these are dev dependencies: 
```
[project.optional-dependencies]
dev = ["ruff>=0.0.254", "pytest", "pytest-xdist"]
```

Reply:
Oh good point; if you want you can add e.g.
`codecarbon = ["codecarbon"]`
so it can be installed via mteb[codecarbon]
or you can also just remove the package requirement and just in the code enclose the `import codecarbon` in a try & except and if it excepts raise an Exception and ask the user to install codecarbon. I like that second option most I think!

Reply:
Okay looks great, I'll do that

Reply:
Well, I liked both options, is it okay to keep them both?

Reply:
Seems fine! Feel free to merge :)